File: python/getNthFib.py

def getNthFib( n ):
  lastTwo = [ 0, 1 ]
  counter = 3
  while counter <= n :
    nextNum = lastTwo[0] + lastTwo[1]
    lastTwo[0] = lastTwo[1]
    lastTwo[1] = nextNum
    counter += 1

  return lastTwo[1] if n > 1 else lastTwo[0]

print(getNthFib(5))

File: go/getNthFib.go
package main

import (
 "fmt"
)

// func main(){
//   fmt.Println(GetNthFib(5))
// }

func GetNthFib( n int ) int{
  lastTwo := []int{0,1}
  counter :=  3
  for counter <= n {
    nextnum := lastTwo[0] + lastTwo[1]
    lastTwo = []int{lastTwo[1],nextnum} 
    counter += 1
   }

  if n > 1 {
    return lastTwo[1]
  }
  return  lastTwo[0]
}

